### ISSUE TITLE
Feature request: handle merge conflicts

### ISSUE THREAD

**Issue by Nadrieril** [rust-analyzer/rust-analyzer#7127]
A big source of annoyance for me is the flurry of error messages when I am resolving a merge conflict. There are syntax errors everywhere which make the code hard to read. Could rust-analyzer notice merge conflicts and refrain from emitting too many syntax errors? Maybe just a single error "syntax error: this is a merge conflict".
More far-fetched would be if ra could actually process both versions in parallel (or all 3 of them in 3-way merges), but I won't hold my breath x)

**Comment by GopherJ** [comment]
I think it's not on rust-analyzer side, vscode can have a `resolve confolicts` mode which disable temporarily diagnostics

**Comment by JohnCSimon** [comment]
What do other IDEs do?

**Comment by gilescope** [comment]
Would be kinda cool though.

**Comment by gilescope** [comment]
The cheapest thing to do would be to use the first block and skip the second block.
That way the rest of the file would parse nicely and you would avoid the ugly sea of red.

That's probably a lot cheaper to implement and would make all the difference. (maybe have a setting to say always take the first or the second conflict chunk?)

**Comment by Nadrieril** [comment]
I note that rustc detects merge conflicts nicely. I don't have an example on hand but it says something like "this is a merge conflict marker, one version of the code is below here, the other is above there"

**Comment by Veykril** [comment]
I agree we should detect this to some degree at least, rustc PR that introduced this for inspiration: https://github.com/rust-lang/rust/pull/106242

**Comment by Jesse-Bakker** [comment]
One feature related to handling merge conflicts that would save me a lot of headaches would be to resolve a merge conflict in imports by merging the respective imports. For example, this is a very common conflict, where one commit added `B` to the imports and the other commit added `C`:
```
<<<<<< HEAD
use crate::{A, B};
======
use crate::{A, C};
>>>>>>
```
In this case, an assist to merge these would be awesome.

**Comment by ChayimFriedman2** [comment]
@Jesse-Bakker Technically, this can be achieved by accepting both and configuring rustfmt import granularity. Though I agree an assist would be nice.
